src/main/java/org/springframework/data/redis/connection/lettuce/LettuceClientConfiguration.java

```diff
@@ -157,6 +163,7 @@ static LettuceClientConfiguration defaultConfiguration() {
 		@Nullable ReadFrom readFrom;
 		Duration timeout = Duration.ofSeconds(RedisURI.DEFAULT_TIMEOUT);
 		Duration shutdownTimeout = Duration.ofMillis(100);
+		Duration shutdownQuietPeriod = Duration.ofMillis(100);
```

Comment:
Let's pre-initialize this configuration parameter with `null` so we can retain the current behavior. `LettuceClientConfiguration.build()` can then check, whether `shutdownQuietPeriod` was configured. If it was configured, we take the value. If it was left unconfigured, we can apply the `shutdownTimeout` as the quiet period.

The shutdown timeout is right now in use for both parameters. If we change the behavior, then all users that expect a near-zero shutdown are forced to change their code.